astropy/cosmology/_src/flrw/base.py

```diff
@@ -308,7 +316,7 @@ def Odm0(self) -> float:
         return self.Om0 - self.Ob0
 
     @cached_property
-    def Ok0(self) -> float:
+    def Ok0(self) -> float:  # type: ignore[override]
```

Comment:
Why do you need that comment?

Reply:
Because this is a read only forever and underlying annotation is a read or write parameter. Subclasses are free to have Ok0 be a dataclass field.

Reply:
Isn't that an actual type error?

Reply:
As far as I can tell this is not some Mypy limitation. It is instead specified by [PEP 544 – Protocols: Structural subtyping (static duck typing)](https://peps.python.org/pep-0544/#protocol-members).

Reply:
But I think the interpretation of how that affects subclasses is checker dependent.

Reply:
If the `Protocol` says the attribute must be writable and it is read-only in the implementing class then the class does not implement the `Protocol`.

Reply:
Depends what one means by implements. Runtime type checkers check for the existence of the attribute, which this satisfies. And statically different checkers have different opinions. The underlying issue is that python itself does not have a means to describe the readability of an attribute without specifying the writability.

Reply:
Do you know if this has been discussed somewhere?

Reply:
The question here is if a `Protocol` defines some variable as writable then does that mean it _must_ be writable or does that mean it _may_ be writable. Apparently Mypy thinks this means it _must_ be writable, but it is worth investigating what other type checkers think. I don't want to make an uninformed decision.

Reply:
Ok. Let's just go with read-only. Subclasses can overwrite it to be readable and it won't complain, but the default is read-only, which I'd rather not do.